pkg/alertmanager/multitenant.go

```diff
@@ -551,6 +592,8 @@ func (am *MultitenantAlertmanager) loadAndSyncConfigs(ctx context.Context, syncR
 	}
 
 	am.syncConfigs(cfgs)
+	am.deleteUnusedLocalUserState()
```

Comment:
You've added a unit test on `deleteUnusedLocalUserState()` (good!). I think would be great if we could improve `TestMultitenantAlertmanager_loadAndSyncConfigs()` to check for deletions too.